diff --git a/cludo_search.routing.yml b/cludo_search.routing.yml
index d91e294..b175742 100644
--- a/cludo_search.routing.yml
+++ b/cludo_search.routing.yml
@@ -9,6 +9,6 @@ cludo_search.search:
   path: '/csearch'
   defaults:
     _form: '\Drupal\cludo_search\Form\CludoSearch'
-    _title: 'Site Search'
+    _title: 'Search the Intranet'
   requirements:
     _permission: 'access cludo search content'
diff --git a/src/Form/CludoSearch.php b/src/Form/CludoSearch.php
index c936038..74ae493 100644
--- a/src/Form/CludoSearch.php
+++ b/src/Form/CludoSearch.php
@@ -25,7 +25,7 @@ class CludoSearch extends FormBase {
   public function buildForm(array $form, FormStateInterface $form_state) {
 
     // Build form.
-    $prompt = $this->t('Enter the terms you wish to search for.');
+    $prompt = '';
     $query = '';
 
     // Basic search.
