diff --git a/core/lib/Drupal/Core/File/FileSystem.php b/core/lib/Drupal/Core/File/FileSystem.php
index be1d70ad6e2..7f389ba082e 100644
--- a/core/lib/Drupal/Core/File/FileSystem.php
+++ b/core/lib/Drupal/Core/File/FileSystem.php
@@ -101,6 +101,12 @@ public function chmod($uri, $mode = NULL) {
       }
     }

+    // We may get forced ogu+rwx permissions from Azure Blob Storage, consider this a success.
+    $actual_perms = substr(sprintf('%o', fileperms($uri)), -3);
+    if (str_starts_with($uri, 'public:/') && $actual_perms == '777') {
+       return TRUE;
+    }
+
     return @chmod($uri, $mode);
   }